lib_nn/api/Filter2D.hpp

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+// #include <cstdint>
+// #include <cstring>
+
+#include "AbstractKernel.hpp"
+#include "AggregateFn.hpp"
+#include "MemCpyFn.hpp"
+#include "OutputTransformFn.hpp"
+#include "Utils.hpp"
+#include "geom/util.hpp"
+
+namespace nn {
+
+/**
+ * Base class for non-depthwise 2D filter kernels.
+ *
+ * This class implements `calc_output_pixel_slice()` used by `AbstractKernel`,
+ * and its behavior is ultimately determined by 3 component objects supplied to
+ * it, the patch handler (instance of `MemCpyFn`), the aggregation handler
+ * (instance of `AggregateFn`) and the output transformer (instance of
+ * `OutputTransformFn`).
+ */
+class Filter2D : public AbstractKernel {
+ public:
+  static constexpr bool UsesPerGroupMemCopy = false;
```

Comment:
What do we use the static variable for?

Reply:
I think it's vestigial and can be safely removed.  Doesn't seem to be used anywhere now.

I know I'm the one that added those in there, though I no longer recall exactly why.  It might have been so that the same unit test somewhere could be used for both `Filter2D` and `Filter2D_DW`.

Reply:
it looks like its used in test_Filter2d.cpp:96
  const auto mult_memcopy = TypeParam::UsesPerGroupMemCopy;

Reply:
improved comment